source/wp-content/themes/wporg-developer-2023/src/code-comment-form/block.php

```diff
@@ -0,0 +1,90 @@
+<?php
+namespace WordPressdotorg\Theme\Developer_2023\Dynamic_Code_Comment_Form;
+
+use function DevHub\can_user_post_note;
+
+add_action( 'init', __NAMESPACE__ . '\init' );
+add_filter( 'comment_form_submit_field', __NAMESPACE__ . '\modify_submit_field', 10, 2 );
+
+/**
+ * Registers the block using the metadata loaded from the `block.json` file.
+ * Behind the scenes, it registers also all assets so they can be enqueued
+ * through the block editor in the corresponding context.
+ *
+ * @see https://developer.wordpress.org/reference/functions/register_block_type/
+ */
+function init() {
+	register_block_type(
+		dirname( dirname( __DIR__ ) ) . '/build/code-comment-form',
+		array(
+			'render_callback' => __NAMESPACE__ . '\render',
+		)
+	);
+}
+
+/**
+ * Render the block content.
+ *
+ * @return string Returns the block markup.
+ */
+function render( $attributes, $content, $block ) {
+	if ( ! isset( $block->context['postId'] ) ) {
+		return '';
+	}
+
+	if ( ! comments_open() || ! can_user_post_note( true, get_the_ID() ) ) {
+		return '';
+	}
+
+	if ( is_user_logged_in() ) {
+		ob_start(); // Capture form output
+		$args = \DevHub_User_Submitted_Content::comment_form_args();
+		comment_form( $args );
+		$form = ob_get_clean(); // End capture
+
+		$editor_rules = \DevHub_User_Submitted_Content::get_editor_rules();
+		$output = "<div class='wp-block-wporg-code-reference-comment-form-content'>{$form}<div>{$editor_rules}</div></div>";
+
+	} else {
+		$output = '<p>' . sprintf(
+			/* translators: %s: login URL */
+			__( 'You must <a href="%s">log in</a> before being able to contribute a note or feedback.', 'wporg' ),
+			esc_url('https://login.wordpress.org/?redirect_to=' . urlencode( get_permalink() ) )
```

Comment:
Could you use [wp_login_url](https://developer.wordpress.org/reference/functions/wp_login_url/) here? It's not _that_ important, but it would help in local testing.

Reply:
I copied this over from the old code because trying to use `wp_login_url` wasn't working. I can't remember why. 

I'll take another look.